src/main/java/TransactionDao.java

```diff
@@ -0,0 +1,105 @@
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+class TransactionDao {
+
+
+    public void addTransaction(Transaction transaction) {
+        Connection connection = connect();
+
+        PreparedStatement preparedStatement = null;
+        try {
+            String sql = "INSERT INTO transactions(type, description, amount, dateStamp) VALUES (?, ?, ?, ?)";
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setString(1, transaction.getTransactionType());
+            preparedStatement.setString(2, transaction.getDescription());
+            preparedStatement.setBigDecimal(3, transaction.getAmount());
+            preparedStatement.setDate(4, Date.valueOf(transaction.getDateStamp()));
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            System.out.println("Brak zapisu do bazy: " + e.getMessage());
+        }
+    }
+
+    public void modifyTransaction(Transaction transaction) {
+        Connection connection = connect();
+
+        PreparedStatement preparedStatement = null;
+        try {
+            String sql = "UPDATE transactions SET transactionType = ?, description = ?, amount = ?, dateStamp = ?) WHERE id = ?";
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setString(1, transaction.getTransactionType());
+            preparedStatement.setString(2, transaction.getDescription());
+            preparedStatement.setBigDecimal(3, transaction.getAmount());
+            preparedStatement.setDate(4, Date.valueOf(transaction.getDateStamp()));
+            preparedStatement.setInt(5, transaction.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            System.out.println("Brak zapisu do bazy: " + e.getMessage());
+        }
+    }
+
+    public void deleteTransaction(Transaction transaction) {
+        Connection connection = connect();
+
+        PreparedStatement preparedStatement = null;
+        try {
+            String sql = "DELETE FROM transactions WHERE id = ?";
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setInt(1, transaction.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            System.out.println("Brak zapisu do bazy: " + e.getMessage());
+        }
+    }
+
+    ResultSet resultSet;
+
+    public List<Transaction> outcomes = new ArrayList<>() {
```

Comment:
Nie za bardzo rozumiem ten kawałek kodu, szczerze mówiąc :P

Tutaj raczej chcesz stworzyć metodę podobną do add/modify, tylko, że twoim sql będzie SELECT. Jak uruchomisz tego selecta, to on zwróci Ci resultSet - i wtedy dopiero będziesz mógł po tym resultSet przeiterować. Zobacz sobie w tej lekcji jak taki SELECT jest robiony: https://javastart.pl/kurs/technologie-webowe/web-sql-bezpieczenstwo/lekcja/web-sql-injection